core/types/receipt_test.go

```diff
@@ -509,43 +511,47 @@ func TestTypedReceiptEncodingDecoding(t *testing.T) {
 	}
 }
 
-func clearComputedFieldsOnReceipts(t *testing.T, receipts Receipts) {
+func clearComputedFieldsOnReceipts(t *testing.T, receipts Receipts) Receipts {
 	t.Helper()
-
-	for _, receipt := range receipts {
-		clearComputedFieldsOnReceipt(t, receipt)
+	r := make(Receipts, len(receipts))
+	for i, receipt := range receipts {
+		r[i] = clearComputedFieldsOnReceipt(t, receipt)
 	}
+	return r
 }
 
-func clearComputedFieldsOnReceipt(t *testing.T, receipt *Receipt) {
+func clearComputedFieldsOnReceipt(t *testing.T, receipt *Receipt) *Receipt {
 	t.Helper()
-
-	receipt.TxHash = libcommon.Hash{}
-	receipt.BlockHash = libcommon.Hash{}
-	receipt.BlockNumber = big.NewInt(math.MaxUint32)
-	receipt.TransactionIndex = math.MaxUint32
-	receipt.ContractAddress = libcommon.Address{}
-	receipt.GasUsed = 0
-
-	clearComputedFieldsOnLogs(t, receipt.Logs)
+	cpy := *receipt
+
+	cpy.TxHash = libcommon.Hash{}
+	cpy.BlockHash = libcommon.Hash{}
+	cpy.BlockNumber = big.NewInt(math.MaxUint32)
+	cpy.TransactionIndex = math.MaxUint32
+	cpy.ContractAddress = libcommon.Address{}
+	cpy.GasUsed = 0
+	cpy.Logs = clearComputedFieldsOnLogs(t, receipt.Logs)
+	return &cpy
```

Comment:
indeed. it need to add optimism derived fields like `L1GasPrice`, `FeeScalar`.
but not about `DepositNonce` because it is not recovered from derive functions.

Reply:
resolved in [0a84e27](https://github.com/testinprod-io/op-erigon/pull/209/commits/0a84e27fbf6425c34a0a1dd9446673fce4fd8544)